### ISSUE TITLE
Camera state update on camera changing issue

### ISSUE THREAD

**Issue by akovalov** [mapbox/mapbox-maps-ios#775]
## Environment

- Xcode version: 13
- iOS version: 15
- Devices affected: iPad
- Maps SDK Version: v10.0.1

## Observed behavior and steps to reproduce

Change map camera by pinch and pan gestures.
The `.cameraChanged` event is reported.
Print camera state on every reported event. Printed state will switch between new correct state and old state at the moment of gesture start.

```
mapView.mapboxMap.onEvery(.cameraChanged) { _ in
            print(self.mapView.mapboxMap.cameraState)
        }
```

The print out will be similar to this. See that first state is equal to third state when second state is correct and it will go like that: correct value, then old value, then again correct and so on.
```
MapboxMaps.CameraState(center: __C.CLLocationCoordinate2D(latitude: 48.67914111569259, longitude: 34.77753011541918), padding: __C.UIEdgeInsets(top: 100.0, left: 100.0, bottom: 100.0, right: 100.0), zoom: 5.762091022431934, bearing: -0.0, pitch: 0.0)
MapboxMaps.CameraState(center: __C.CLLocationCoordinate2D(latitude: 46.83731384346994, longitude: 45.52068366519899), padding: __C.UIEdgeInsets(top: 100.0, left: 100.0, bottom: 100.0, right: 100.0), zoom: 5.762091022431934, bearing: -0.0, pitch: 0.0)
MapboxMaps.CameraState(center: __C.CLLocationCoordinate2D(latitude: 48.67914111569259, longitude: 34.77753011541918), padding: __C.UIEdgeInsets(top: 100.0, left: 100.0, bottom: 100.0, right: 100.0), zoom: 5.762091022431934, bearing: -0.0, pitch: 0.0)
```

## Expected behavior

self.mapView.mapboxMap.cameraState will return a new correct (currently visible map) state on every event reported.

**Comment by dandyweng** [comment]
Any updates? I’m trying to implement a feature based on cameraState changes and I would love to see this bug fixed as soon as possible.

**Comment by patrickkempff** [comment]
I can confirm we have the same issue when trying to implement a feature based on the zoom level of the camera. @ZiZasaurus

**Comment by macdrevx** [comment]
We're working on some changes that should address this.

**Comment by macdrevx** [comment]
I'm reopening this issue because we've discovered that the solution we implemented in https://github.com/mapbox/mapbox-maps-ios/pull/837 introduces a new issue described in https://github.com/mapbox/mapbox-maps-ios/issues/864.

For 10.2, developers will need to make their own tradeoff between these two issues using the new configuration option introduced in https://github.com/mapbox/mapbox-maps-ios/pull/890.

In the future, we hope to eliminate the need to make this tradeoff, so we will keep this ticket open until then.

cc: @ohjonah @akovalov @dandyweng @patrickkempff

**Comment by macdrevx** [comment]
Was closed inadvertently.

**Comment by naftalibeder** [comment]
This does not appear to be fixed in 10.9.0. Is this a known regression? @macdrevx

**Comment by macdrevx** [comment]
cc: @ZiZasaurus
